extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTable.java

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.engine.table.Table;
+import org.apache.iceberg.Snapshot;
+import org.jetbrains.annotations.NotNull;
+
+public interface IcebergTable extends Table {
+    /**
+     * Update the table with the latest snapshot from the catalog.
+     */
+    @SuppressWarnings("unused")
+    void update();
```

Comment:
If we're going to want this method, it suggests that we need to be able to expose the current `Snapshot` and/or ID to users via this interface, so they can avoid hitting an ISE in the other overloads.